### PR TITLE
Fix for WFWIP-439 and WFWIP-440, provisioning-dir can be set to absolute path, '.' and outside target dir

### PR DESCRIPTION
**Pull request by jfdenise**
Issues:
https://issues.redhat.com/browse/WFWIP-439
https://issues.redhat.com/browse/WFWIP-440

### CODE REVIEW

**Comment by jamezp** [comment]
@jfdenise I don't follow how those two JIRA's related to this change.

**Comment by jfdenise** [comment]
@jamezp , thank-you I made a typo in the JIRA number. It has been fixed.

File: plugin/src/main/java/org/wildfly/plugin/provision/AbstractProvisionServerMojo.java [hunk 1]
```diff
@@ -157,9 +157,9 @@
     private File provisioningFile;
 
     /**
-     * The target directory the application to be deployed is located.
+     * The target directory. Read-only.
      */
-    @Parameter(defaultValue = "${project.build.directory}/", property = PropertyNames.DEPLOYMENT_TARGET_DIR)
+    @Parameter(defaultValue = "${project.build.directory}/", readonly = true)
     protected File targetDir;
```
Comment (jamezp) [inline plugin/src/main/java/org/wildfly/plugin/provision/AbstractProvisionServerMojo.java:163]: Not really a big deal, but you could just remove this and then below using `project.getBuild().getDirectory()`.

Comment (jfdenise) [reply plugin/src/main/java/org/wildfly/plugin/provision/AbstractProvisionServerMojo.java]: @jamezp , thank-you. Fixed.

**Review by jamezp — APPROVED** [review]

**Pull request merged**